File: type-ui/src/main/java/dev/sayaya/handbook/client/interfaces/api/LayoutRepository.java
package dev.sayaya.handbook.client.interfaces.api;

import dev.sayaya.handbook.domain.LayoutPeriod;
import dev.sayaya.handbook.domain.Position;
import dev.sayaya.handbook.domain.TypeLayout;
import dev.sayaya.rx.Observable;

import java.util.List;
import java.util.Map;

public interface LayoutRepository {
    Observable<List<TypeLayout>> layouts();
    Observable<Map<String, Position>> positions(LayoutPeriod period);
    Observable<Void> savePositions(LayoutPeriod period, Map<String, Position> positions);
}

File: persist-type/src/main/kotlin/dev/sayaya/handbook/usecase/LayoutRepository.kt
package dev.sayaya.handbook.usecase

import dev.sayaya.handbook.domain.TypeLayout
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono
import java.util.*

/**
 * 타입 레이아웃 영속화 포트 (헥사고날 아키텍처 출력 포트).
 *
 * **책임:** 워크스페이스별 타입 캔버스 레이아웃(노드 위치)의 조회 및 저장을 정의한다.
 *
 * **의존관계:**
 * - [R2dbcLayoutRepositoryAdapter][dev.sayaya.handbook.interfaces.database.R2dbcLayoutRepositoryAdapter] — R2DBC 구현체
 */
interface LayoutRepository {
    fun findByWorkspace(workspace: UUID): Flux<TypeLayout>
    fun save(workspace: UUID, layout: TypeLayout): Mono<TypeLayout>
}
